airflow/providers/amazon/aws/sensors/sagemaker_training.py

```diff
@@ -72,6 +75,15 @@ def get_sagemaker_response(self):
         if self.print_log:
             if not self.log_resource_inited:
                 self.init_log_resource(self.get_hook())
+            if (
+                self.instance_count is None
+                or self.state is None
+                or self.last_description is None
+                or self.last_describe_job_call is None
+            ):
+                raise AirflowException(
+                    "instance_count, state, last_description, last_describe_job_call must be specified"
```

Comment:
Are you sure about this?

If this is the case, we should raise this in the constructor so it is a DAG parse time error, rather than a run/execution time error

Reply:
I think it is better to specify the initial value of the class rather than generating an error.

```
self.instance_count = 1
self.state = LogState.STARTING
self.last_description = {}
self.last_describe_job_call = 0
self.log_resource_inited = False
```